File: combinatorics/resource/ST/project_5/bai_toan_4/mutil_graph/amap_to_al.cpp
#include <iostream>
#include <vector>
#include <tuple>

using namespace std;

// Ham chuyen AMap sang AL
vector<vector<int>> amapToAl(int n, const vector<tuple<int, int, int>>& amap_entries) {
    vector<vector<int>> al(n);

    for (auto [u, v, w] : amap_entries) {
        for (int i = 0; i < w; ++i)
            al[u].push_back(v);  // Them v vao danh sach ke cua u w lan
    }

    return al;
}

int main() {
    int n, k;
    cout << "Nhap so dinh: ";
    cin >> n;
    cout << "Nhap so bo (u v w): ";
    cin >> k;

    vector<tuple<int, int, int>> amap_entries;

    cout << "Nhap cac bo (u v w):\n";
    for (int i = 0; i < k; ++i) {
        int u, v, w;
        cin >> u >> v >> w;
        amap_entries.emplace_back(u, v, w);
    }

    auto al = amapToAl(n, amap_entries);

    cout << "Danh sach ke (AL):\n";
    for (int u = 0; u < n; ++u) {
        cout << u << ": ";
        for (int v : al[u])
            cout << v << " ";
        cout << "\n";
    }

    return 0;
}

File: combinatorics/resource/ST/project_5/bai_toan_4/mutil_graph/amap_to_al.py
def amap_to_al(n, amap_entries):
    # Khoi tao danh sach ke rong
    al = [[] for _ in range(n)]

    for u, v, w in amap_entries:
        al[u].extend([v] * w)  # Them v vao danh sach ke cua u w lan

    return al

# Nhap tu ban phim
n = int(input("Nhap so dinh: "))
k = int(input("Nhap so bo (u v w): "))

print("Nhap cac bo (u v w):")
amap_entries = []
for _ in range(k):
    u, v, w = map(int, input().split())
    amap_entries.append((u, v, w))

al = amap_to_al(n, amap_entries)

print("Danh sach ke (AL):")
for u in range(n):
    print(f"{u}: {' '.join(map(str, al[u]))}")